pkg/engine/template_generator.go

```diff
@@ -209,6 +209,9 @@ func (t *TemplateGenerator) getTemplateFuncMap(cs *api.ContainerService) templat
 		"IsHostedMaster": func() bool {
 			return cs.Properties.IsHostedMasterProfile()
 		},
+		"IsHostedMasterIPMasqAgentDisabled": func() bool {
```

Comment:
oh, do not know that, just to confirm, as you said, it will have the context of cs.Properties in template and call the correct IsHostedMasterIPMasqAgentDisabled() method in kubernetesagentcustomdata.yml?

Reply:
Nevermind I saw you use this in `kubernetesagentcustomdata.yml` which has context for the agentpool profile but not properties.